elastic/beats
# Set the fileProspector's ignoreInactiveSince value

The filestream input's fileProspector does not take the configured ignore_inactive option into account. This option could be especially important if the registry is not persisted before Filebeat is restarted. In this case, setting ignore_inactive to since_last_start should cause Filebeat to only read the changes made since the restart and ignore old files, but since this value is not actually set in the fileProspector, every file will be fully read again.



## What does this PR do?

Sets the ignoreInactiveSince option on the fileProspector, based on the ignore_inactive configuration option.



## Why is it important?

Because the ignore_inactive option is currently ignored.


## Checklist




## How to test this PR locally
Start filebeat with the ignore_inactive option set to since_last_start. Example:
```
filebeat.inputs:
- type: filestream
  id: json-log-files
  ignore_inactive: since_last_start
  paths:
    - ./data/logs/*.json
  parsers:
    - ndjson:
        message_key: message
      
output.file:
  path: ./data/filebeat_output
```
Add your log files. The contents of the log files are added to the output.
Stop filebeat, and remove the registry.
Start filebeat, and add new content to the log files.
Expected: Only the newly added content is added to the output.
Result without the fix: The full content of the log files was added to the output.
Result with the fix: Only the newly added content is added to the output.
